gatorgrade/input/command_line_generator.py

```diff
@@ -0,0 +1,66 @@
+"""Generates a dictionary of shell and gator grader command options from a list of dict checks."""
+
+import os
+
+
+# Function to generate command options from a list of object checks
+
+# pylint: disable=too-many-nested-blocks
+def generate_checks(file_context_checks):
+    """Generate a dictionary of checks based on the configuration file.
+
+        This dictionary will have the format:
+        {
+            "shell": List of shell checks,
+            "gatorgrader": List of GatorGrader checks
+        },
+    Args:
+        file_context_checks: List containing dictionaries that contain file contexts
+            (either a file path or None if no file context) and checks in another dictionary
+            (can be either GatorGrader or shell checks).
+            The input list is generated based on the configuration file.
+    """
+    gatorgrader_checks = []
+    shell_checks = []
+    for file_context_check in file_context_checks:
+        # assigning the check from the dict object
+        check = file_context_check["check"]
+        # If the check has a 'command', then it is a shell check
+        if "command" in check:
+            shell_checks.append(check)
+        # Else it's a gator grader check
+        else:
+            gatorgrader_command_options = []
+            # Defining the description and option
+            description = check["description"]
+            options = check["options"]
+            # Creating a list that has description, check, and options in it for GatorGrader check
+            gatorgrader_command_options = ["--description", f"{description}"]
+            gatorgrader_command_options.append(check["check"])
+            # If options exist add all the keys and the values into GatorGrader command options
+            if options:
+                for key in options:
+                    # If the type of the value is boolean, add the key but not the boolean value
+                    # Checking if the key is a flag
```

Comment:
These two comments kind of say the same thing but in different ways -- either combining them, or just removing one will make this clearer and easier to read.

Reply:
Sounds good!